src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -236,11 +236,11 @@ protected AnalyzeResponse shardOperation(AnalyzeRequest request, ShardId shardId
 
             int position = 0;
             while (stream.incrementToken()) {
+                tokens.add(new AnalyzeResponse.AnalyzeToken(term.toString(), position, offset.startOffset(), offset.endOffset(), type.type()));
                 int increment = posIncr.getPositionIncrement();
                 if (increment > 0) {
                     position = position + increment;
                 }
-                tokens.add(new AnalyzeResponse.AnalyzeToken(term.toString(), position, offset.startOffset(), offset.endOffset(), type.type()));
```

Comment:
This doesn't look correct to me: `of course` would be analyzer as a single token `course` at position 0 while it should be at position 1.

I think the right fix is to put back `tokens.add` where it was before and instead to initialize `position` at -1.